### ISSUE TITLE
Request of the tensorflow model "cifar10_model.pb" 

### ISSUE THREAD

**Issue by pkiourti** [YiZeng623/frequency-backdoor#1]
Hi,

Thank you for sharing the frozen graph of the backdoored model cifar10_model.pb. 

Is it possible to upload the TensorFlow model in another format so that we can access its layers? I tried translating the frozen graph to a ".h5" file. There is not enough information to do so (there is an error about "No variables to save", and to overcome this, I would need access to the TensorFlow variables).

Thank you so much,
Penny

**Comment by YiZeng623** [comment]
Here you go. I have uploaded the .h5 file under the same folder. Enjoy.

**Comment by pkiourti** [comment]
Using the model you shared, the smooth trigger and its dominant label (i.e., 6), poisoning 10% of the retraining, should give 95% ASR from the 1st iteration as mentioned in the paper (Sec 5.3). 

Of course, I don't have the details of how you retrained it, but I chose to retrain the model you shared using SGD and typical learning rate, and **I can achieve an accuracy of 85%, but the ASR is very low in my case (< 25%)**. 

Is it possible to share the actual backdoored model that responds to this smooth trigger so that we can properly refer to/cite your method?

**Comment by YiZeng623** [comment]
Hi, thanks for the update. Just to clarify, our claim was that we train a **new model from scratch** (in your case, you mentioned the term retrain, which could be the problem) using the poisoned dataset (smooth trigger polluting 10% of non-target-class samples + dominant label (which should be computed across different datasets)).

**Comment by YiZeng623** [comment]
If the problem still remains, could you please share the smooth trigger you generated with me?

**Comment by YiZeng623** [comment]
You can also take a look into the test set that you used to acquire the ASR, whose label should all be set to 6 if you R attacking using the target label '6'.

**Comment by pkiourti** [comment]
Thank you for your quick response. I am using the smooth trigger you provide under data/best_universal.npy, and I poison 10% of the training non-target images with the trigger and set their label to 6. 

*I was indeed retraining the model that you provided. I made 2 changes, and I now get ASR 98% and Acc 83%:*
1) I trained a new model from scratch (which I believe needs to use the same architecture as the one used to produce the smooth trigger?) 
2) I copied the part of the code that adds the trigger to one image at a time from target_universal.py and created a function out of it. I think the batch poisoning I was doing before, was done incorrectly (in terms of the final clipped values maybe?) even though I had tested it through jupyter notebook. Although I am not sure if that was the issue, I believe retraining should not have been a problem based on the backdoor literature. 

Thank you for your help,
Penny

**Comment by YiZeng623** [comment]
You R welcomed. 
So, for 1, not necessarily, you can choose whatever model you like and it should all work. As long as you conduct the backdoor attack using the Cifar10 dataset, which is the dataset you used to generate the smooth trigger. The mode during smooth trigger generation step simply provides gradients to find perturbations;
For 2, normally not a problem. However, U should not use the same training dataset as the poisoned dataset (I assume you adopt the pre-trained weight from cifar10 to fit with the new poison dataset, this will diminish the attack efficacy). A possible case in the community is that you use pre-trained weights from another dataset, say ImageNet, then you use this weight to transfer learning over a cifar10 poisoned dataset. You can give this case a shot, and with more epochs to train I guess. But, this case was not evaluated in our paper, so, thanks for mentioning that.
